beaker/client/application_client.py

```diff
@@ -70,24 +71,53 @@ def compile(
         return (b64decode(result["result"]), result["hash"], src_map)
 
     def build(self):
+        """Wraps the Application in an AppPrecompile before handing off to
+        _build_program for recursive compiling. The result is then used
+        to assign the approval and clear state program binaries and src maps."""
+        app_precompile = AppPrecompile(self.app)
+        compiled_app = self._build_program(app_precompile)
```

Comment:
This will cause a rebuild maybe more frequently than we want, probably not really an issue but the previous one checked to see if the binaries were avail before trying to compile them

Reply:
Added back the checks